pylint/lint/pylinter.py

```diff
@@ -1442,7 +1444,10 @@ def _get_message_state_scope(
         """Returns the scope at which a message was enabled/disabled."""
         if confidence is None:
             confidence = interfaces.UNDEFINED
-        if self.config.confidence and confidence.name not in self.config.confidence:
+        if (
+            self.namespace.confidence
+            and confidence.name not in self.namespace.confidence
+        ):
             return MSG_STATE_CONFIDENCE  # type: ignore[return-value] # mypy does not infer Literal correctly
```

Comment:
Look like we handling the case where ``self.config.confidence`` is falsey, but it should be None or a confidence, at this point, right ? Is that for mypy ?

Reply:
That's interesting actually. I'll try and see if this could just be an `in` check.